### PR TITLE
KAFKA-12905: Replace EasyMock and PowerMock with Mockito for NamedCacheMetricsTest

### PR DESCRIPTION
**Pull request by wycccccc**
Development of EasyMock and PowerMock has stagnated while Mockito continues to be actively developed. With the new Java cadence, it's a problem to depend on libraries that do bytecode generation and are not actively maintained. In addition, Mockito is also easier to [use.KAFKA-7438](https://issues.apache.org/jira/browse/KAFKA-7438?jql=project%20%3D%20KAFKA%20AND%20status%20in%20(Open%2C%20Reopened)%20AND%20assignee%20in%20(EMPTY)%20AND%20text%20~%20%22mockito%22)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 1]
```diff
@@ -0,0 +1,2 @@
+mock-maker-inline
```
Comment (chia7712) [inline streams/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: Why we need this file?

**Review by chia7712 — COMMENTED** [review]
@wycccccc thanks for your contribution. Could you revise the title? "KAFKA-12905"

File: build.gradle [hunk 2]
```diff
@@ -1710,6 +1710,7 @@ project(':streams') {
     testImplementation libs.powermockEasymock
```
Comment (ijuma) [inline build.gradle:1710]: Can we remove the powermock dependencies or are there other tests in Streams using it?

File: streams/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 1 above)
Comment (ijuma) [reply streams/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: I don't think we need this although if we use mocking of statics, we would need to have a dependency on mockito-inline vs mockito-core (like we do in `tools`).

File: build.gradle (hunk 2 above)
Comment (wycccccc) [reply build.gradle:1710]: There are other dependencies that have not been modified, so this cannot be removed .

File: streams/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 1 above)
Comment (wycccccc) [reply streams/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: Thanks suggestion, I have modified.

File: build.gradle [hunk 3]
```diff
@@ -1484,6 +1484,7 @@ project(':storage') {
     testImplementation project(':clients').sourceSets.test.output
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
+    testImplementation libs.mockitoInline // supports mocking static methods, final classes, etc.
```
Comment (ijuma) [inline build.gradle]: Why have we made this change?

File: build.gradle [hunk 4]
```diff
@@ -1710,6 +1711,8 @@ project(':streams') {
     testImplementation libs.powermockEasymock
     testImplementation libs.bcpkix
     testImplementation libs.hamcrest
+    testImplementation libs.mockitoCore
```
Comment (ijuma) [inline build.gradle]: We probably don't need this, right?

File: build.gradle (hunk 3 above)
Comment (wycccccc) [reply build.gradle]: sor,my mis.

File: build.gradle (hunk 4 above)
Comment (wycccccc) [reply build.gradle]: Yes,It's extra.I have removed.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/NamedCacheMetricsTest.java [hunk 5]
```diff
@@ -49,37 +41,28 @@
...
-            HIT_RATIO_MAX_DESCRIPTION);
-        replay(streamsMetrics);
-        replay(StreamsMetricsImpl.class);
+                expectedSensor,
+                StreamsMetricsImpl.CACHE_LEVEL_GROUP,
+                tagMap,
+                hitRatio,
+                HIT_RATIO_AVG_DESCRIPTION,
+                HIT_RATIO_MIN_DESCRIPTION,
+                HIT_RATIO_MAX_DESCRIPTION);
```
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/NamedCacheMetricsTest.java]: nit
```suggestion
            expectedSensor,
            StreamsMetricsImpl.CACHE_LEVEL_GROUP,
            tagMap,
            hitRatio,
            HIT_RATIO_AVG_DESCRIPTION,
            HIT_RATIO_MIN_DESCRIPTION,
            HIT_RATIO_MAX_DESCRIPTION
        );
```

**Review by cadonna — APPROVED** [review]
@wycccccc Thank you for the PR!

I have only a nit indentation request.

If @ijuma is fine with the changes in `build.gradle`, I can merge.

**Comment by ijuma** [comment]
@cadonna The build changes look good.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/NamedCacheMetricsTest.java [hunk 6]
```diff
@@ -49,37 +41,28 @@
     private static final String HIT_RATIO_MIN_DESCRIPTION = "The minimum cache hit ratio";
     private static final String HIT_RATIO_MAX_DESCRIPTION = "The maximum cache hit ratio";
 
-    private final StreamsMetricsImpl streamsMetrics = createMock(StreamsMetricsImpl.class);
-    private final Sensor expectedSensor = mock(Sensor.class);
+    private Sensor expectedSensor = Mockito.mock(Sensor.class);
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/NamedCacheMetricsTest.java]: How about adding ‘final’ back?

**Review by chia7712 — APPROVED** [review]
one trivial comment below. Otherwise LGTM

File: streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/NamedCacheMetricsTest.java (hunk 6 above)
Comment (wycccccc) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/NamedCacheMetricsTest.java]: OK,I have changed.Thanks your suggestion.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/NamedCacheMetricsTest.java (hunk 5 above)
Comment (wycccccc) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/NamedCacheMetricsTest.java]: Thank you for the review. @ijuma  @chia7712 @cadonna

**Comment by cadonna** [comment]
Test failures are unrelated and known to be flaky:
```
Build / JDK 8 and Scala 2.12 / kafka.server.RaftClusterTest.testCreateClusterAndCreateAndManyTopicsWithManyPartitions()
Build / JDK 15 and Scala 2.13 / kafka.server.RaftClusterTest.testCreateClusterAndCreateAndManyTopics()
```

**Pull request merged**